src/main.cpp

```diff
@@ -3342,6 +3342,13 @@ bool static DisconnectTip(CValidationState &state, const CChainParams &chainpara
     if (!ReadBlockFromDisk(block, pindexDelete, chainparams.GetConsensus()))
         return AbortNode(state, "Failed to read block");
 
+    // retrieve all mints
+    block.sigmaTxInfo = std::make_shared<sigma::CSigmaTxInfo>();
+    for (auto const& tx : block.vtx) {
+        CheckTransaction(tx, state, tx.GetHash(), false, pindexDelete->pprev->nHeight,
```

Comment:
Why we need to call CheckTransaction() here?

Reply:
CheckTransaction provide logic to retrieve mint and serial if we pass sigmaTxInfo with. Also I need to use same logic to retrieve instead of retrieve it manually.

Reply:
What you mean retrieve?

Reply:
actually, just get mint/spend list from block.

Reply:
Ok got it.